jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -64,8 +64,8 @@ public class ArrayByteBufferPool implements ByteBufferPool, Dumpable
     private final int _maxCapacity;
     private final long _maxHeapMemory;
     private final long _maxDirectMemory;
-    private final AtomicLong _currentHeapMemory = new AtomicLong();
-    private final AtomicLong _currentDirectMemory = new AtomicLong();
+    private final AtomicLong _heapMemory = new AtomicLong();
+    private final AtomicLong _directMemory = new AtomicLong();
```

Comment:
These are not kept up to date for leaked buffers

Reply:
Yes they are.

Reply:
Can you point to the mechanism? I couldn't see it.

Reply:
I can see how you've added `leaked()` to the `ConcurrentPool` as I did, but I can't see where that is used to decrement these sizes?

Also as @lorban pointed out in in my PR, this doesn't handle leaks from the `QueuePool`

Reply:
It does handle leaks. The decrement is done in acquire, not release, as explained in previous comments.

If a buffer is acquired, the memory is decremented as now the buffer is in use, and if it is leaked, it will be eventually GC'ed, but its memory has already been accounted for.

Reply:
Ah I get it now.  You've changed the definition of the pool size to count only idle buffers.     I can see this makes sense... however, it is a little strange that a ConcurrentPool can be full of in-use entries, but by this metric it will be considered empty.     I guess that is why we default to the compound one.   More responses in the other PR....